services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/UpdateHankeITests.kt

```diff
@@ -599,9 +601,18 @@ class UpdateHankeITests(
         val createdHanke = hankeFactory.builder(USERNAME).withHankealue().save()
         val hankealue = createdHanke.alueet[0]
         assertThat(hankealue.nimi).isEqualTo("Hankealue 1")
+        val haittojenhallintasuunnitelma = createHaittojenhallintasuunnitelma()
         val request = createdHanke.toModifyRequest()
         val updatedRequest =
-            request.copy(alueet = listOf(request.alueet[0].copy(nimi = "Changed Name")))
+            request.copy(
+                alueet =
+                    listOf(
+                        request.alueet[0].copy(
+                            nimi = "Changed Name",
+                            haittojenhallintasuunnitelma = haittojenhallintasuunnitelma
```

Comment:
I think there should be tests where the content of one value in the map is updated and where a single value is removed and added. If we only test the none or all situations, it feels like there could be bugs remaining.

Reply:
I added more tests.